src/services/search/listingSearch.ts

```diff
@@ -39,27 +44,83 @@ export const fetchListingCount = async ({
   }
 }
 
-export const defaultSort = {
+export const fetchDealerListingsCount = async ({
+  dealerId,
+  query = {},
+  options = {},
+}: {
+  dealerId: number
+  query?: ListingSearchParams
+  options?: ApiCallOptions
+}): Promise<number> => {
+  const { count } = await postData({
+    path: `dealers/${dealerId}/listings/count`,
+    body: {
+      query,
+    },
+    options: {
+      isAuthorizedRequest: true,
+      ...options,
+    },
+  })
+
+  return count
+}
+
+export const fetchDealerArchivedListingsCount = async ({
+  dealerId,
+  options,
+}: {
+  dealerId: number
+  options?: ApiCallOptions
+}): Promise<number> => {
+  const { count } = await fetchPath({
+    path: `dealers/${dealerId}/archived-listings/count`,
+    options: {
+      isAuthorizedRequest: true,
+      ...options,
+    },
+  })
+
+  return count
+}
+
+export const defaultUserSort = {
   sortType: ListingSortTypeParams.RELEVANCE,
   sortOrder: ListingSortOrderParams.ASC,
 }
 
-const defaultPagination = {
+export const defaultDealerSort = {
+  sortType: ListingSortTypeParams.CREATED_DATE,
+  sortOrder: ListingSortOrderParams.DESC,
+}
+
+export const defaultUserPagination = {
   page: 0,
   size: 24,
 }
 
+export const defaultDealerPagination = {
```

Comment:
What's the reason to have a different default pagination length for the User and the Dealer?

Reply:
Gold listing